### PR TITLE
Update result type templates

### PR DESCRIPTION
**Pull request by reekitconcept**
Support all preview images

- preview image for news
- for all content types, if exist
- use volto's Image component

- add compatibility for Volto 16 (or less)

Update default icon

### CODE REVIEW

File: src/components/theme/SolrSearch/resultItems/helpers/ResultItemPreviewImage.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import { Link } from 'react-router-dom';
+import { flattenToAppURL } from '@plone/volto/helpers';
+
+// XXX Note that the indexing must be fixed instead!
+const fixJson = (json) => json.replace(/'/g, '"');
+
+// Note this could be done via Image content type, but it's simpler and more effective
+// to provide a direct link and avoid srcsets altogether.
```
[the remark this answers is not in the document]
Comment (reekitconcept) [reply src/components/theme/SolrSearch/resultItems/helpers/ResultItemPreviewImage.jsx:8]: Simply put, if we use srcsets: the downside is that the browser might still download the full size image, for example if it decides to do so before the css arrives even. So it starts downloading without knowing the final size. We realized this during testing DLR. The sscset implementations have changed substantially from its first draft from years agoe, and now there is no way to hint the browser to do anything else that it finds the most efficient.

This means that if we do know that the image size is a thumbnail, the most efficient way is still to load the image in the required thumbnail size. (In fact my component has a maxwidth property that we could use to cap the size in this case, but I don't know if the new core volto compoment supports this..)

Another, independent remark. You have noticed that I'm trying to convert the json format to a standard json format. However we have fixed this at some point, so now there should be a serializer that produces the correct json format. If I remember correctly. So right now it seems to me that the field value is broken.

[the remark this answers is not in the document]
Comment (davisagli) [reply src/components/theme/SolrSearch/resultItems/helpers/ResultItemPreviewImage.jsx:8]: @reekitconcept The volto Image component supports a `sizes` attribute to specify the layout width of the image:
- https://6.docs.plone.org/volto/recipes/images.html#serving-the-right-image-size
- https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/sizes

> Another, independent remark. You have noticed that I'm trying to convert the json format to a standard json format. However we have fixed this at some point, so now there should be a serializer that produces the correct json format. If I remember correctly. So right now it seems to me that the field value is broken.

The actual field value is a Python structure, not serialized (well, serialized as a pickle, obviously). This should be working correctly if the field was indexed with collective.solr 9.0.1 or later, which has this fix: https://github.com/collective/collective.solr/pull/347 -- double check that you don't have old data in the index.

[the remark this answers is not in the document]
Comment (reekitconcept) [reply src/components/theme/SolrSearch/resultItems/helpers/ResultItemPreviewImage.jsx]: @datisagli I've amended the PR to use the Image component from Volto.

The PR is still WIP with 2 outstanding problems:

- serialization. FYI I'm testing this on BFS. I'll check why I use the wrong version.
- icons. as I mentioned they still need an overhaul.

**Comment by reekitconcept** [comment]
Ok the icons will go to a separate PR, so this is ready to go.

^ @davisagli

**Review by davisagli — COMMENTED** [review]
@reekitconcept The tests are failing because they run on Volto 16, and the Image component was only added in recent Volto 17 alpha versions.

I think we could probably update the tests to use Volto 17, since it is already used by the current projects where we plan to use this addon.

But, if we also need to support Volto 16, then we need a fallback to be used in case the Image component was not found in the registry.

**Comment by reekitconcept** [comment]
> @reekitconcept The tests are failing because they run on Volto 16, and the Image component was only added in recent Volto 17 alpha versions.
> 
> I think we could probably update the tests to use Volto 17, since it is already used by the current projects where we plan to use this addon.
> 
> But, if we also need to support Volto 16, then we need a fallback to be used in case the Image component was not found in the registry.

Thank you! I'll take care of that tomorrow.

I still have the old version that does not use the image component so I can put that back as a fallback. Should be simple. I think it's worth to support Volto 16 if this is the only reason against it.

**Comment by davisagli** [comment]
@reekitconcept Yeah, sounds good

**Pull request merged**